File: src/main/java/org/tindalos/principle/domain/checker/DesignQualityCheckService.java
package org.tindalos.principle.domain.checker;

import java.util.List;

import org.tindalos.principle.domain.core.DesignQualityCheckConfiguration;
import org.tindalos.principle.domain.core.Package;

public class DesignQualityCheckService {

    private final PackageAnalyzer packageAnalyzer;
    private final DesignQualityDetectorsRunner designQualityDetectorsRunner;

    public DesignQualityCheckService(PackageAnalyzer packageAnalyzer, DesignQualityDetectorsRunner designQualityDetectorsRunner) {
        this.packageAnalyzer = packageAnalyzer;
        this.designQualityDetectorsRunner = designQualityDetectorsRunner;
    }

    public DesignQualityCheckResults analyze(DesignQualityCheckConfiguration parameters) {
        List<Package> packages = packageAnalyzer.analyze(parameters);
        return designQualityDetectorsRunner.execute(packages, parameters);
    }

}

File: src/main/java/org/tindalos/principle/domain/checker/DesignQualityCheckService.scala
package org.tindalos.principle.domain.checker

import org.tindalos.principle.domain.core.DesignQualityCheckConfiguration
import scala.collection.JavaConversions._
import org.tindalos.principle.domain.core.Package

class DesignQualityCheckService(val packageAnalyzer:PackageAnalyzer
								, val designQualityDetectorsRunner: DesignQualityDetectorsRunner) {
  
    def analyze(parameters: DesignQualityCheckConfiguration) = {
        val packages = packageAnalyzer.analyze(parameters)
        designQualityDetectorsRunner.detectorResults(packages, parameters)
    }

}